packages/nimble-components/CONTRIBUTING.md

```diff
@@ -490,12 +490,13 @@ To modify the generated tokens, complete these steps:
 
 Public names for theme-aware tokens are specified in `src/theme-provider/design-token-names.ts`. Use the following structure when creating new tokens.
 
-`[element]-[part]-[state]-[token_type]`
+`[element]-[part]-[state]-[variant]-[token_type]`
 
 1. Where **element** is the type to which the token applies (e.g. 'application', 'body', or 'title-plus-1').
 2. Where **part** is the specific part of the element to which the token applies (e.g. 'border', 'background', or shadow).
-3. Where **state** is the more specific state descriptor (e.g. 'selected' or 'disabled'). Multiple states should be sorted alphabetically.
-4. Where **token_type** is the token category (e.g. 'color', 'font', 'font-color', 'height', 'width', or 'size').
+3. Where **state** is one or more of the following interaction states: 'active', 'disabled', 'hover', or 'selected'. Multiple states should be sorted alphabetically.
+4. Where **variant** is any other distinguishing descriptor (e.g. 'accent', 'primary, or 'large'). Multiple variants should be sorted alphabetically.
```

Comment:
"state" vs "variant" / "distinguishing descriptor" does not have a clear concept to me. i.e. it's not clear what it is a variant of. The examples are more like variants of the element/part not variants of the interaction state. If it's modelled after the appearance variant concept it's clearer that in that context it's a variant of appearance.

I'd propose `interaction state` vs `appearance state`. I think the examples show how it's not over constrained to the appearance attribute concepts (i.e. by including 'large')

As an aside this kind of also parallels the [cascade guidance](https://github.com/ni/nimble/blob/main/packages/nimble-components/docs/css-guidelines.md#prefer-cascade-for-state-changes) where the interaction states for the most part are before other states.

Reply:
My intent was for "variant" to be a catch-all for however else we're distingushing this token from another (with the same element, part, and/or interaction state). It may not be an appearance-based distinction. As a contrived example, suppose our token is a distinct width or font property that we want to use when the locale is German (because German words tend to be longer and take up more space). That doesn't seem like an "appearance state" to me.

As a counter proposal, I'm changing these to `interaction_state` and `other_state`, but I'm happy to use something else if you don't like `other_state`.

Reply:
gave an alternative and approved assuming we are aligned, if not can give a counter